src/mongocxx/test/spec/initial_dns_seedlist_discovery.cpp

```diff
@@ -0,0 +1,320 @@
+
+    auto admin = client["admin"];
+    auto result = admin.run_command(make_document(kvp("ping", 1)));
+    REQUIRE(result.begin()->get_double() == 1.0);
+
+    auto uri = client.uri();
+    auto my_options = uri.options();
+    auto creds = uri.credentials();
+    compare_options(test.options, my_options, creds);
+
+    bool has_hosts_list = !test.hosts.empty();
+    if (test.num_hosts || has_hosts_list) {
+        size_t count = 0;
+        while (!hosts_are_equal(test.hosts, new_hosts, mtx)) {
+            count++;
+            if (count > 4) {
+                REQUIRE(hosts_are_equal(test.hosts, new_hosts, mtx));
+            }
+            std::this_thread::sleep_for(std::chrono::seconds(count));
+        }
+    }
+
+    if (test.num_hosts) {
+        REQUIRE((size_t)test.num_hosts == new_hosts.size());
+    }
+}
+
+static void run_srv_max_hosts_test_file(const InitialDNSSeedlistTest& test) {
+    using namespace mongocxx;
+
+    bool should_ping = true;
+    if (!test.ping) {
+        should_ping = false;
+    }
+    should_ping = should_ping && !test.error;
+
+    options::apm apm_opts;
+    std::mutex mtx;
+    std::vector<std::string> new_hosts;
+
+    apm_opts.on_topology_changed([&](const events::topology_changed_event& event) {
+        const std::lock_guard<std::mutex> lock(mtx);
+        auto new_td = event.new_description();
+        auto servers = new_td.servers();
+        new_hosts.clear();
+        for (const auto& i : servers) {
+            new_hosts.push_back(std::string(i.host()) + ":" + std::to_string(i.port()));
+        }
+    });
+
+    mongocxx::options::tls tls_options;
+    mongocxx::options::client client_options;
+
+    auto ca_file_path = std::getenv("MONGOCXX_TEST_TLS_CA_FILE");
+    REQUIRE(ca_file_path);
+    tls_options.ca_file(ca_file_path);
+    tls_options.allow_invalid_certificates(true);
+    client_options.tls_opts(tls_options);
+    client_options.apm_opts(apm_opts);
+
+    try {
+        mongocxx::uri my_uri{test.uri};
+        class client client {
+            my_uri, test_util::add_test_server_api(client_options)
+        };
+        bool using_tls = client.uri().tls();
+        if (!using_tls) {
+            return;
+        }
+        REQUIRE(!test.error);
+        if (should_ping) {
+            validate_srv_max_hosts(client, test, mtx, new_hosts);
+            mongocxx::pool pool{my_uri,
+                                options::pool(test_util::add_test_server_api(client_options))};
+            auto pool_client = pool.acquire();
+            validate_srv_max_hosts(*pool_client, test, mtx, new_hosts);
+        }
+    } catch (mongocxx::exception& e) {
+        REQUIRE(test.error);
+    }
+}
+
+static void iterate_srv_max_hosts_tests(std::string dir, std::vector<std::string> files) {
+    for (const auto& file : files) {
+        auto test_doc = _doc_from_file("/" + dir + "/" + file);
+        auto test = InitialDNSSeedlistTest::parse(test_doc);
+        run_srv_max_hosts_test_file(test);
```

Comment:
```suggestion
        SECTION (file) {
            run_srv_max_hosts_test_file(test);
        }
```

Suggest adding a section for file names to enable running one test file:

```
./cmake-build/src/mongocxx/test/test_driver  "uri::test_srv_max_hosts" --section "replica-set" --section "srvMaxHosts-less_than_srv_records.json"
```